File: models/doctormanagement/DoctorOperational.js
const DoctorOperationalSchema = new mongoose.Schema({
    hospitalId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      index: true
    },
  
    doctorId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
      index: true
    },
  
    // ✅ Branch-wise Schedule
    schedule: [{
      branchId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        index: true
      },
      dayOfWeek: { type: Number, required: true }, // 0-6
      startTime: String,
      endTime: String,
      slotDuration: { type: Number, default: 15 }
    }],
  
    // ✅ Leave (no need branch here normally)
    leaves: [{
      fromDate: Date,
      toDate: Date,
      reason: String,
      status: {
        type: String,
        enum: ["Pending", "Approved", "Rejected"],
        default: "Pending"
      }
    }],
  
    // ✅ Branch-wise Shift
    shifts: [{
      branchId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true
      },
      shiftName: String,
      startTime: String,
      endTime: String
    }],
  
    // ✅ Daily Appointment Limit
    maxAppointmentsPerDay: {
      type: Number,
      default: 50
    }
  
  }, { timestamps: true });
  
  export const DoctorOperational =
    mongoose.model("DoctorOperational", DoctorOperationalSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/DoctorOperational.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "doctor_operationals")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DoctorOperational extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", nullable = false)
    private User doctor;

    @ElementCollection
    @CollectionTable(name = "doctor_schedules", joinColumns = @JoinColumn(name = "doctor_operational_id"))
    private List<Schedule> schedule;

    @ElementCollection
    @CollectionTable(name = "doctor_leaves", joinColumns = @JoinColumn(name = "doctor_operational_id"))
    private List<LeaveItem> leaves;

    @ElementCollection
    @CollectionTable(name = "doctor_shifts", joinColumns = @JoinColumn(name = "doctor_operational_id"))
    private List<ShiftItem> shifts;

    @Column(name = "max_appointments_per_day")
    private Integer maxAppointmentsPerDay = 50;

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class Schedule {
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "branch_id", nullable = false)
        private Branch branch;

        @Column(name = "day_of_week", nullable = false)
        private Integer dayOfWeek; // 0-6

        @Column(name = "start_time")
        private String startTime;
        @Column(name = "end_time")
        private String endTime;
        @Column(name = "slot_duration")
        private Integer slotDuration = 15;
    }

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class LeaveItem {
        @Column(name = "from_date")
        private LocalDateTime fromDate;
        @Column(name = "to_date")
        private LocalDateTime toDate;
        private String reason;
        @Enumerated(EnumType.STRING)
        private LeaveStatus status = LeaveStatus.Pending;
    }

    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class ShiftItem {
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "branch_id", nullable = false)
        private Branch branch;

        @Column(name = "shift_name")
        private String shiftName;
        @Column(name = "start_time")
        private String startTime;
        @Column(name = "end_time")
        private String endTime;
    }

    public enum LeaveStatus {
        Pending, Approved, Rejected
    }
}
